tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,544 @@
+
+            // Update values in the store
+            await _configClient.SetConfigurationSettingAsync(new ConfigurationSetting($"{keyPrefix}:Setting1", "UpdatedValue1"));
+            await _configClient.SetConfigurationSettingAsync(new ConfigurationSetting($"{keyPrefix}:Setting2", "UpdatedValue2"));
+            await _configClient.SetConfigurationSettingAsync(new ConfigurationSetting(sentinelKey, "Updated"));
+
+            // Wait for cache to expire
+            await Task.Delay(TimeSpan.FromSeconds(2));
+
+            // Act
+            var result = await refresher.TryRefreshAsync();
+
+            // Assert
+            Assert.True(result);
+            Assert.Equal("UpdatedValue1", config[$"{keyPrefix}:Setting1"]);
+            Assert.Equal("InitialValue2", config[$"{keyPrefix}:Setting2"]); // This value shouldn't change
+        }
+
+        [Fact]
+        public async Task TryRefreshAsync_RefreshesFeatureFlags_WhenConfigured()
+        {
+            Skip.If(_skipTests, _skipReason);
+
+            // Arrange - Setup test-specific keys
+            var (keyPrefix, sentinelKey, featureFlagKey) = await SetupTestKeys("RefreshesFeatureFlags");
+            IConfigurationRefresher refresher = null;
+
+            var config = new ConfigurationBuilder()
+                .AddAzureAppConfiguration(options =>
+                {
+                    options.Connect(GetConnectionString());
+                    options.Select($"{keyPrefix}:*");
+                    options.UseFeatureFlags();
+
+                    options.ConfigureRefresh(refresh =>
+                    {
+                        refresh.Register(sentinelKey)
+                              .SetRefreshInterval(TimeSpan.FromSeconds(1));
+                    });
+
+                    refresher = options.GetRefresher();
+                })
+                .Build();
+
+            // Verify initial feature flag state
+            Assert.Equal("False", config[$"FeatureManagement:{keyPrefix}Feature:Enabled"]);
+
+            // Update feature flag in the store
+            await _configClient.SetConfigurationSettingAsync(
+                ConfigurationModelFactory.ConfigurationSetting(
+                    featureFlagKey,
+                    @"{""id"":""" + keyPrefix + @"Feature"",""description"":""Test feature"",""enabled"":true}",
+                    contentType: FeatureManagementConstants.ContentType));
+            await _configClient.SetConfigurationSettingAsync(new ConfigurationSetting(sentinelKey, "Updated"));
+
+            // Wait for cache to expire
+            await Task.Delay(TimeSpan.FromSeconds(2));
+
+            // Act
+            var result = await refresher.TryRefreshAsync();
+
+            // Assert
+            Assert.True(result);
+            Assert.Equal("True", config[$"FeatureManagement:{keyPrefix}Feature:Enabled"]);
+        }
+
+        [Fact]
+        public async Task RegisterAll_RefreshesAllKeys_WhenSentinelChanged()
```

Comment:
sentinel and refresh all are two completely different paths. I don't see a reason to mix them.